pkg/satellite/stream/stream.go

```diff
@@ -181,7 +180,32 @@ func (ss *satelliteStream) ackReceivedTelemetry(telemetryMessageAckId string) {
 	}
 }
 
+func (ss *satelliteStream) performAutoClose() {
+	log.Printf("Stream auto-close conditions met - exiting")
+	if ss.showStats {
+		metrics.logReport()
+	}
+	ss.Close()
+	os.Exit(0)
+}
+
 func (ss *satelliteStream) recvLoop() {
+	streamEndDetected := false
+
+	if ss.enableAutoClose {
+		ticker := time.NewTicker(1 * time.Second)
+		go func() {
+			for {
+				select {
+				case <-ticker.C:
+					if streamEndDetected {
+						ss.performAutoClose()
+					}
+				}
```

Comment:
```suggestion
				<-ticker.C
				if streamEndDetected {
					ss.performAutoClose()
				}
```

Reply:
Now that there is only one `case`, a `select` statement may no longer be necessary.